### DISCUSSION TITLE
Refresh token

### DISCUSSION THREAD

**Discussion by Vette1123** [nextauthjs/next-auth#10397] [Help]
Can you kindly add a live example of how to integrate refresh token logic into V5?, or any updated example of how to apply refresh token or updating the session in next.js 13+++?

**Comment by PAAPII10** [comment]
# Using `unstable_update` in NextAuth Beta

You can use the `unstable_update` helper from NextAuth to update the session server-side.

## 1. Import `unstable_update`

```ts
export const { handlers, signIn, signOut, auth, unstable_update } = NextAuth({
  // your config
});
```
## 2. Create a server function to update the session
Inside a `"use server"` file, define a helper like this:
```ts
"use server";

import { auth, unstable_update } from "@/auth"; // adjust path accordingly
import type { Session } from "next-auth";

export const updateSession = async ({
  data,
}: {
  data: Partial<Session["user"]>;
}) => {
  const session = await auth();
  if (session) {
    await unstable_update({
      ...session,
      user: {
        ...session.user,
        ...data,
      },
    });
  }
};
```

## 3. Call updateSession wherever you need to refresh/update session data
For example, after refreshing access tokens:

```ts
const newToken = await arvasitAuth.refreshAccessToken({
  token: refreshToken,
});

if (newToken.accessToken) {
  setCookie("accessToken", newToken.accessToken);
  setCookie("refreshToken", newToken.refreshToken);

  await updateSession({
    data: {
      accessToken: newToken.accessToken,
      refreshToken: newToken.refreshToken,
    },
  });
}
```
